pink/tasks/relative_frame_task.py

```diff
@@ -109,15 +113,19 @@ def set_orientation_cost(
         """
         if isinstance(orientation_cost, float):
             assert orientation_cost >= 0.0
-        else:  # not isinstance(orientation_cost, float)
-            assert all(cost >= 0.0 for cost in orientation_cost)
-        if isinstance(self.cost, np.ndarray):
-            self.cost[3:6] = orientation_cost
-        else:  # self.cost is not a vector
-            raise TaskDefinitionError(
-                "Frame task cost should be a vector, "
-                f"currently cost={self.cost}"
-            )
+        else:  # Should be a ndarray or seq
+            if not isinstance(orientation_cost, np.ndarray):  # Must be seq
+                try:
+                    orientation_cost = np.array(orientation_cost)
+                except Exception:  # Not a proper float sequence
+                    raise TaskDefinitionError(
+                        "Orientation task cost should be a float or a "
+                        "seq of float or ndarray of size 1 or 3,"
+                        f"currently cost={self.cost}"
+                    )
+                assert np.all(orientation_cost >= 0.0)
```

Comment:
Same comment as above. With this level of indentation we won't check cost positivity in the event where the vector was already an NDArray.

Reply:
I think it is better to set at the end, to avoid setting negative value if the user use a `try`/`except`.